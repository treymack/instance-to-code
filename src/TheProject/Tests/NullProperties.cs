using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests;
using NUnit.Framework;

namespace TheProject.Tests
{
    [TestFixture]
    public class NullProperties
    {
        [Test]
        public void CanScriptToCode()
        {
            var instance = new NullPropertiesTestCase();

            var code = Instance.ToCode(instance);

            Approvals.Verify(code);
        }
    }

    public class NullPropertiesTestCase
    {
        public DateTime? NullableDateTime { get; set; }
        public string String1 { get; set; }
        public NullPropertiesTestCase2 TestCase2 { get; set; }
    }
  
    public class NullPropertiesTestCase2
    {
    }
}
