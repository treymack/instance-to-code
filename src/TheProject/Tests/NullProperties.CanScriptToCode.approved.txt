new TheProject.Tests.NullPropertiesTestCase()
{
NullableDateTime = null,
String1 = null,
TestCase2 = null,
}